sigstore/_cli.py

```diff
@@ -536,10 +547,26 @@ def _verify(args: argparse.Namespace) -> None:
     elif args.rekor_url == DEFAULT_REKOR_URL:
         verifier = Verifier.production()
     else:
-        # TODO: We need CLI flags that allow the user to figure the Fulcio cert chain
-        # for verification.
-        args._parser.error(
-            "Custom Rekor and Fulcio configuration for verification isn't fully supported yet!",
+        if not args.certificate_chain:
+            args._parser.error(
+                "Custom Rekor URL used without specifying --certificate-chain"
+            )
+
+        try:
+            certificate_chain = _split_certificate_chain(
+                args.certificate_chain.read_text()
+            )
+        except SplitCertificateChainError as error:
+            args._parser.error(f"Failed to split certificate chain: {error}")
+
+        verifier = Verifier(
+            rekor=RekorClient(
+                url=args.rekor_url,
+                pubkey=args.rekor_root_pubkey.read(),
```

Comment:
I wonder whether we should warn if a user supplies `--rekor-root-pubkey` but don't set the `--rekor-url`. Same with the new `--certificate-chain`. At the moment they just get ignored which might be confusing.

This issue also exists in the signing code path.

Reply:
Yeah, it would be good to catch this and flag it as a warning at the minimum (or maybe even an error, since we expect it to fail anyways).

Reply:
Let's track this with a follow-up issue.